lib/inspec/attribute_registry.rb

```diff
@@ -0,0 +1,71 @@
+require 'forwardable'
+require 'singleton'
+require 'inspec/objects/attribute'
+
+module Inspec
+  class AttributeRegistry
+    include Singleton
+    extend Forwardable
+
+    attr_reader :list
+    def_delegator :list, :each
+    def_delegator :list, :[]
+    def_delegator :list, :key?, :profile_exist?
+    def_delegator :list, :select
+
+    def self.find_attribute(name, profile)
```

Comment:
hrrrrmmmm  If this is a singleton, I'd expect to call these via 

```
Inspec::AttributeRegistry.instance.find_attribute(...)```

I suppose the class methods are just a convenience? Might add a comment to that effect.